modules/firstPartyData/index.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * This module sets default values and validates ortb2 first part data
+ * @module modules/firstPartyData
+ */
+
+import { config } from '../../src/config.js';
+import * as utils from '../../src/utils.js';
+import { getHook } from '../../src/hook.js';
+import { getRefererInfo } from '../../src/refererDetection.js'
+import { getStorageManager } from '../../src/storageManager.js';
+
+const STORAGE = getStorageManager();
+let ortb2 = {};
+let globalConfig = {};
+let win = (window === window.top) ? window : window.top;
+
+/**
+ * Checks for referer and if exists merges into ortb2 global data
+ */
+function setReferer() {
+  if (getRefererInfo().referer) utils.mergeDeep(ortb2, {site: {ref: getRefererInfo().referer}});
+}
+
+/**
+ * Checks for canonical url and if exists merges into ortb2 global data
+ */
+function setPage() {
+  if (getRefererInfo().canonicalUrl) utils.mergeDeep(ortb2, {site: {page: getRefererInfo().canonicalUrl}});
+}
+
+/**
+ * Checks for canonical url and if exists retrieves domain and merges into ortb2 global data
+ */
+function setDomain() {
+  let parseDomain = function(url) {
+    if (!url || typeof url !== 'string' || url.length === 0) return;
+
+    var match = url.match(/^(?:https?:\/\/)?(?:www\.)?(.*?(?=(\?|\#|\/|$)))/i);
+
+    return match && match[1];
+  };
+
+  let domain = parseDomain(getRefererInfo().canonicalUrl)
+
+  if (domain) utils.mergeDeep(ortb2, {site: {domain: domain}});
+}
+
+/**
+ * Checks for screen/device width and height and sets dimensions
+ */
+function setDimensions() {
+  let width;
+  let height;
+
+  try {
+    width = win.innerWidth || win.document.documentElement.clientWidth || win.document.body.clientWidth;
+    height = win.innerHeight || win.document.documentElement.clientHeight || win.document.body.clientHeight;
+  } catch (e) {
+    width = window.innerWidth || window.document.documentElement.clientWidth || window.document.body.clientWidth;
+    height = window.innerHeight || window.document.documentElement.clientHeight || window.document.body.clientHeight;
+  }
+
+  utils.mergeDeep(ortb2, {device: {width: width, height: height}});
```

Comment:
The spec [here](https://github.com/prebid/Prebid.js/issues/6099#issue-759814143) specifies `ortb2.device.h` and `ortb2.device.w` but here, `ortb2.device.height` and `ortb2.device.width` is used.

Reply:
Fixing this. Thanks for catching!